File: javastory.budgetsh.stage4/src/javastory/budgetsh/stage4/client/ui/budget/menu/TransactionMenu.java
package javastory.budgetsh.stage4.client.ui.budget.menu;

import java.util.Scanner;

import javastory.budgetsh.stage4.client.dto.budget.CashBook;
import javastory.budgetsh.stage4.client.ui.budget.console.TransactionConsole;
import javastory.budgetsh.stage4.client.util.Narrator;
import javastory.budgetsh.stage4.client.util.TalkingAt;

public class TransactionMenu {
	//
	private TransactionConsole transactionConsole;
	private CashBook currentBook;
	
	private Scanner scanner;
	private Narrator narrator;
	
	public TransactionMenu() {
		//
		transactionConsole = new TransactionConsole();
		
		scanner = new Scanner(System.in);
		narrator = new Narrator(this, TalkingAt.Left);
	}
	
	public void show(CashBook currentBook) {
		//
		if(currentBook!=null) {
			this.currentBook = currentBook;
		}
		
		int inputNumber = 0;
		
		while(true) {
			//
			displayMenu();
			inputNumber = selectMenu(); 
			
			switch (inputNumber) {
			//
			case 1:
				transactionConsole.register(this.currentBook);
				break;
			case 2:
				transactionConsole.find(this.currentBook);
				break;
			case 3:
				transactionConsole.findAll(this.currentBook);
				break;
			case 4:
				transactionConsole.modify(this.currentBook);
				break;
			case 5:
				transactionConsole.remove(this.currentBook);
				break;
			case 6:
				this.currentBook = transactionConsole.selectBook();
				break;
			case 0:
				return;
				
			default:
				narrator.sayln("choose again!");
				continue;
			}
		}
	}
	
	public void displayMenu() {
		//
		narrator.sayln("------------------------------");
		if (currentBook!=null) {
			narrator.sayln(" Transaction Menu [ " + currentBook.getTravel().getName()+" ]");
		}else {
			narrator.sayln(" Transaction Menu [ No selected ]");
		}
		narrator.sayln("------------------------------");
		narrator.sayln(" 1. register");
		narrator.sayln(" 2. find");
		narrator.sayln(" 3. findAll");
		narrator.sayln(" 4. modify");
		narrator.sayln(" 5. remove");
		narrator.sayln(" 6. select CashBook");
		narrator.sayln("------------------------------");
		narrator.sayln(" 0. Previous Menu");
	}
	
	public int selectMenu() {
		//
		narrator.say("Select : ");
		int menuNumber = scanner.nextInt();
		
		if (menuNumber >= 0 && menuNumber <= 6) {
			scanner.nextLine();
			return menuNumber;
		} else {
			narrator.sayln(" It's invalid Number -->" + menuNumber);
		}
		return menuNumber;
	}
}

File: javastory.budgetsh.stage5/src/ui/menu/TransactionMenu.js
var co = require('co');
var prompt = require('co-prompt');
const TransactionConsole = require('../console/TransactionConsole.js');
const transactionConsole = new TransactionConsole();

class TransactionMenu {
	constructor() {
		this.currentBook = null;
	}
	show(previousMenu) {
		this.selectMenu(previousMenu);
	}

	displayMenu() {
		console.log("------------------------------");
		console.log('Transaction Menu')
		console.log("------------------------------");
		console.log(" 1. register");
		console.log(" 2. find");
		console.log(" 3. modify");
		console.log(" 4. remove");
		console.log("------------------------------");
		console.log(" 0. Previous Menu");
	}

	selectMenu(previousMenu) {
		var selectedNumber = '';

		this.displayMenu();
		co(function*() {
			var selectedNumber = yield prompt('>  ');
			var numberLine = Number(selectedNumber);
			if (numberLine > 0 && numberLine <= 4) {
				sendToService(numberLine);
			} else if (numberLine == 0) {
				previousMenu.show();
			} else {
				this.selectNumber(previousMenu);
			}
		});
	}

}

var transactionMenu = new TransactionMenu();

function sendToService(inputNumber) {
	//var currentBook = null;
	switch (inputNumber) {
	//
	case 1:
		transactionConsole.register(transactionMenu);
		break;
	case 2:
		transactionConsole.find(transactionMenu);
		break;
	case 3:
		transactionConsole.modify(transactionMenu);
		break;
	case 4:
		transactionConsole.remove(transactionMenu);
		break;
	case 0:
		return;
	}
}

module.exports = TransactionMenu;